### ISSUE TITLE
`let_unit_value` still has a false positive when the expression is an async function with a generic return type

### ISSUE THREAD

**Issue by booleancoercion** [rust-lang/rust-clippy#10455]
### Summary

Similarly to the last comment in #8998 before it was closed, this lint is emitted in error, since the let binding is used to specify the return type in this case. Satisftying the lint by removing the let binding altogether causes a compiler error.

### Lint Name

let_unit_value

### Reproducer

```rust
pub async fn generic<T>() -> T {
    unimplemented!()
}

pub async fn calling_generic() {
    let () = generic().await;
}
```

### Version

```text
Occurs on the playground on both stable 1.67.1 and nightly 1.70.
```

### Additional Labels

@rustbot label +l-suggestion-causes-error

**Comment by Jarcho** [comment]
This was fixed in #10439

**Comment by booleancoercion** [comment]
My bad, that issue was a bit hard to look up
